### PR TITLE
Implements isPlaying() for android and ios

### PR DESCRIPTION
**Pull request by sunjae1294**
- This implements isPlaying() for both android and ios.
- This correctly handles timing issues between `js` and native code by using event emitter.
- This PR reflects comment given in PR  #349 .
 - @trepidity . More features from PR #349 will be added as separate PR as soon as this PR is merged to upstream.

### CODE REVIEW

**Comment by ohtangza** [comment]
@ghsdh3409 Can you review this? Final check.

File: RNSound/RNSound.m [hunk 1]
```diff
@@ -76,6 +76,7 @@ -(void) audioPlayerDidFinishPlaying:(AVAudioPlayer*)player
   if (key == nil) return;
 
   @synchronized(key) {
+    [self setOnPlay:NO forPlayerKey:key];
```
Comment (ghsdh3409) [inline RNSound/RNSound.m]: `audioPlayerDidFinishPlaying` is called when `AVAudioPlayer` pause or stop?

Comment (ghsdh3409) [inline RNSound/RNSound.m]: `audioPlayerDidFinishPlaying` is called when `AVAudioPlayer` pause or stop? If not, it should be added in the `stop` and `pause`.

File: android/src/main/java/com/zmxv/RNSound/RNSoundModule.java [hunk 2]
```diff
@@ -214,6 +223,7 @@ public void play(final Integer key, final Callback callback) {
       @Override
       public synchronized void onCompletion(MediaPlayer mp) {
         if (!mp.isLooping()) {
+          setOnPlay(false, key);
```
Comment (ghsdh3409) [inline android/src/main/java/com/zmxv/RNSound/RNSoundModule.java]: `onCompletion ` is called when `MediaPlayer` pause or stop? If not, it should be added in the `stop` and `pause`.

Comment (ghsdh3409) [reply android/src/main/java/com/zmxv/RNSound/RNSoundModule.java]: Ah! I found that you handle `stop` and `pause` case in the sound.js! Is there any reason that you handle this on sound.js not native code?

Comment (sunjae1294) [reply android/src/main/java/com/zmxv/RNSound/RNSoundModule.java]: `stop` and `pause` has explicit callback function. So handling it in callback function would be faster than using event Emitter!

**Comment by trepidity** [comment]
@sunjae1294 just let me know when you are ready for the merge.

**Comment by sunjae1294** [comment]
@trepidity Thanks! This PR is ready for the merge.

**Comment by ohtangza** [comment]
@sunjae1294 I think this needs some update on README.md. Please let me know if you do not have much time, then I would add myself :)

**Comment by sunjae1294** [comment]
@ohtangza Sorry, I was away for New Year holiday. I'll add README.md and additional features!

**Comment by ohtangza** [comment]
@sunjae1294 Wa~~~ Looking forward to your contribution :) Thanks! (You can work on your personal forked repo, I think!)

**Comment by sunjae1294** [comment]
@ohtangza I actually meant additional features from Hayanmind/react-native-sounds ^^. We still got some other features from Hayanmind/react-native-sounds (prevent auto playing)

**Pull request merged**